PyMca5/tests/ut/test_q_tiled_catalog_selector_dialog.py

```diff
@@ -129,3 +132,80 @@ def test_navigation(
 
     for row_num, text in enumerate(expected_text):
         assert dialog.catalog_table.item(row_num, 0).text() == text
+
+
+def test_current_page_layout(
+        qtbot: QtBot,
+        tiled_client_dialog_model: TiledCatalogSelector
+):
+    """Check current page layout of breadcrumbs updates correctly."""
+    dialog = QTiledCatalogSelectorDialog(model=tiled_client_dialog_model)
+    dialog.show()
+    qtbot.addWidget(dialog)
+
+    dialog.model.enter_node("structured_data")
+    expected_widget_text_order = ["root", " / ", "struc...", " / "]
+    expected_widget_types = [ClickableIndexedQLabel, QLabel]
+
+    for index, expected_widget_type, expected_widget_text in zip(
+        range(dialog.current_path_layout.count()),
+        cycle(expected_widget_types),
+        expected_widget_text_order
+    ):
+        widget = dialog.current_path_layout.itemAt(index).widget()
+        assert isinstance(widget, expected_widget_type)
+        assert widget.text() == expected_widget_text
+
+    dialog.model.exit_node()
+    expected_widget_text_order = ["root", " / "]
+    expected_widget_types = [ClickableIndexedQLabel, QLabel]
+
+    for index, expected_widget_type, expected_widget_text in zip(
+        range(dialog.current_path_layout.count()),
+        cycle(expected_widget_types),
+        expected_widget_text_order
+    ):
+        widget = dialog.current_path_layout.itemAt(index).widget()
+        assert isinstance(widget, expected_widget_type)
+        assert widget.text() == expected_widget_text
+
+    dialog.model.enter_node("structured_data")
+
+    dialog.model.jump_to_node(0)
+    expected_widget_text_order = ["root", " / "]
+    expected_widget_types = [ClickableIndexedQLabel, QLabel]
+
+    for index, expected_widget_type, expected_widget_text in zip(
+        range(dialog.current_path_layout.count()),
+        cycle(expected_widget_types),
+        expected_widget_text_order
+    ):
+        widget = dialog.current_path_layout.itemAt(index).widget()
+        assert isinstance(widget, expected_widget_type)
+        assert widget.text() == expected_widget_text
+
+
+def test_clicking_breadcrumbs(
+        qtbot: QtBot,
+        tiled_client_dialog_model: TiledCatalogSelector
+):
+    """Check clicking breadcrumbs displays correct contents."""
+    dialog = QTiledCatalogSelectorDialog(model=tiled_client_dialog_model)
+    dialog.show()
+    qtbot.addWidget(dialog)
+
+    dialog.model.enter_node("structured_data")
+
+    # Click "struc..." breadcrumb
+    bc_widget = dialog.current_path_layout.itemAt(2).widget()
+    bc_widget.clicked.emit(1)
+    expected_text = ["..", "pets"]
+    for row_num, text in enumerate(expected_text):
+        assert dialog.catalog_table.item(row_num, 0).text() == text
+
+    # Click root breadcrumb
+    dialog.current_path_layout.itemAt(0)
```

Comment:
This line is not used now.  Should it be this instead?

```suggestion
    bc_widget = dialog.current_path_layout.itemAt(0)
```